alacritty/src/event.rs

```diff
@@ -919,13 +919,15 @@ impl<'a, N: Notify + 'a, T: EventListener> ActionContext<'a, N, T> {
     fn update_cursor_blinking(&mut self) {
         // Get config cursor style.
         let mut cursor_style = self.config.terminal_config.cursor.style;
-        if self.terminal.mode().contains(TermMode::VI) {
+        let vi_mode = self.terminal.mode().contains(TermMode::VI);
+        if vi_mode {
             cursor_style = self.config.terminal_config.cursor.vi_mode_style.unwrap_or(cursor_style);
-        };
+        }
 
         // Check terminal cursor style.
         let terminal_blinking = self.terminal.cursor_style().blinking;
-        let blinking = cursor_style.blinking_override().unwrap_or(terminal_blinking);
+        let mut blinking = cursor_style.blinking_override().unwrap_or(terminal_blinking);
+        blinking &= vi_mode || self.terminal().mode().contains(TermMode::SHOW_CURSOR);
```

Comment:
You don't like the following?

```suggestion
        let blinking = cursor_style.blinking_override().unwrap_or(terminal_blinking)
            && (vi_mode || self.terminal().mode().contains(TermMode::SHOW_CURSOR));
```

I'm somewhat open to both here, but I'm surprised you didn't go for the solution that is shorter and doesn't require defining the variable as mutable, or modification of it.

Merge whatever you prefer, but I'd favor the suggestion.

Reply:
I had something like that but it looked  way too long for me, and it has more brackets...

Reply:
I mean it's objectively shorter and it's not like a single set of brackets is much of a deal.